[CI] CrossClusterQueryAllCombinationsSkipUnavailableIT test failing

**Build Scans:**
- [elasticsearch-intake #37215 / part3](https://gradle-enterprise.elastic.co/s/by6vvx7znniyq)
- [elasticsearch-pull-request #135479 / part-3](https://gradle-enterprise.elastic.co/s/yyacgsd4isdmy)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.CrossClusterQueryAllCombinationsSkipUnavailableIT.test" -Dtests.seed=A29DBCC127DFAFBA -Dtests.locale=th-Thai-TH -Dtests.timezone=Arctic/Longyearbyen -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.CrossClusterQueryAllCombinationsSkipUnavailableIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Request breaker not reset to 0 on node: cluster-a-0
Expected: <0L>
     but: was <704L>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test test
- [main] 1 consecutive failures in step part3
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 2 failures in test test (0.7% fail rate in 283 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

It looks like there was an assertion error before releasing circuit-breaker bytes.
I believe this is actual cause:
```
java.lang.AssertionError: elasticsearch[cluster-a-1][transport_worker][T#3] not in [esql_worker, search] nor a test thread	
	at __randomizedtesting.SeedInfo.seed([2968A00442D8AA8D]:0)	
	at org.elasticsearch.threadpool.ThreadPool.assertCurrentThreadPool(ThreadPool.java:1115)	
	at org.elasticsearch.xpack.esql.plugin.DataNodeRequestSender.trySendingRequestsForPendingShards(DataNodeRequestSender.java:199)	
	at org.elasticsearch.xpack.esql.plugin.DataNodeRequestSender$1.onAfterRequest(DataNodeRequestSender.java:295)	
	at org.elasticsearch.xpack.esql.plugin.DataNodeRequestSender$1.onResponse(DataNodeRequestSender.java:313)	
	at org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:274)
```
and `java.lang.AssertionError: Request breaker not reset to 0 on node: cluster-a-0` is only a symptom as bytes were not released after.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -86,9 +86,6 @@ tests:
 - class: org.elasticsearch.xpack.autoscaling.storage.ReactiveStorageIT
   method: testScaleWhileShrinking
   issue: https://github.com/elastic/elasticsearch/issues/145669
-- class: org.elasticsearch.xpack.esql.action.CrossClusterQueryAllCombinationsSkipUnavailableIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/145884
 - class: org.elasticsearch.xpack.esql.expression.function.scalar.multivalue.MvPercentileTests
   method: "testEvaluateBlockWithNulls {TestCase=field: <random mv DEDUPLICATED_UNORDERD doubles>, percentile: <positive int>}"
   issue: https://github.com/elastic/elasticsearch/issues/145886

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/DataNodeRequestSender.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/DataNodeRequestSender.java
@@ -40,6 +40,7 @@ import org.elasticsearch.threadpool.ThreadPool;
 import org.elasticsearch.transport.TransportException;
 import org.elasticsearch.transport.TransportRequestOptions;
 import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.transport.Transports;
 import org.elasticsearch.xpack.esql.action.EsqlSearchShardsAction;
 
 import java.util.ArrayList;
@@ -195,7 +196,8 @@ abstract class DataNodeRequestSender {
     }
 
     private void trySendingRequestsForPendingShards(TargetShards targetShards, ComputeListener computeListener) {
-        assert ThreadPool.assertCurrentThreadPool(ThreadPool.Names.SEARCH);
+        assert ThreadPool.assertCurrentThreadPool(ThreadPool.Names.SEARCH)
+            || (rootTask.isCancelled() && Transports.isTransportThread(Thread.currentThread()));
         changed.set(true);
         final ActionListener<Void> listener = computeListener.acquireAvoid();
         try {

